support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -187,36 +196,111 @@ const stripeAccount = query.product !== 'Contribution' ? 'REGULAR' : 'ONE_OFF';
 const stripePublicKey = getStripeKey(
 	stripeAccount,
 	countryId,
-	currentCurrencyKey,
+	currencyKey,
 	isTestUser,
 );
 
-export function Checkout() {
-	if (!query.product || !query.ratePlan) {
-		return <div>Not enough query parameters</div>;
+/**
+ * Product config - we check that the querystring returns a product, ratePlan and price
+ */
+const productId = query.product in productCatalog ? query.product : undefined;
+const product = productId ? productCatalog[query.product] : undefined;
+const ratePlan = product?.ratePlans[query.ratePlan];
+const price = ratePlan?.pricing[currencyKey];
+const productDescription = productId
+	? productCatalogDescription[productId]
+	: undefined;
+const ratePlanDescription = productDescription?.ratePlans[query.ratePlan];
+
```

Comment:
This was hoisted from inside the Component to outside. We might be able to encapsulate this meaning more meaningfully by adding them as `props` to the component - PR coming soon!